models.ts

```diff
@@ -1,18 +1,54 @@
-import * as jsonfile from "jsonfile";
+import { readFile, writeFile } from "jsonfile";
 
 // no modificar estas propiedades, agregar todas las que quieras
 class Peli {
-  id: number;
-  title: string;
-  tags: string[];
+   id: number;
+   title: string;
+   tags: string[];
 }
 
 class PelisCollection {
-  getAll(): Promise<Peli[]> {
-    return jsonfile("...laRutaDelArchivo").then(() => {
-      // la respuesta de la promesa
-      return [];
-    });
-  }
+   moviesColection: Peli[];
+
+   getAll() {
+      return readFile("./pelis.json").then((res) => {
+         this.moviesColection = res;
+         return this.moviesColection;
+      });
+   }
+   getById(id: number) {
+      return this.getAll().then((res) => {
+         return res.find((p) => p.id == id);
+      });
+   }
+   search(options: any) {
+      return this.getAll().then((res) => {
+         var aux = res;
+
+         if (options.hasOwnProperty("title")) {
+            aux = res.filter((p) => p.title.includes(options.title));
+         }
+         if (options.hasOwnProperty("tag")) {
+            aux = aux.filter((p) => p.tags.includes(options.tag));
+         }
+         return aux;
+      });
+   }
+   add(Peli: Peli) {
+      return this.getAll().then((res) => {
```

Comment:
Esto lo podés simplificar usando this.getById() que hace exactamente lo mismo.

Reply:
Uh tenés razón, lo terminé codeando dos veces y son exactamente iguales. Lo peor es que los pensé y escribí dos veces, ni siquiera es que copié y pegué. Núnca me dí cuenta. Muchisimas gracias por los comentarios Marce, siempre sirven y estoy chocho con este desafío, me pareció super copado